src/coreclr/src/jit/gentree.cpp

```diff
@@ -12498,6 +12498,37 @@ GenTree* Compiler::gtFoldTypeCompare(GenTree* tree)
         objOp = opOther->AsCall()->gtCallThisArg->GetNode();
     }
 
+    bool                 pIsExact   = false;
+    bool                 pIsNonNull = false;
+    CORINFO_CLASS_HANDLE objCls     = gtGetClassHandle(objOp, &pIsExact, &pIsNonNull);
+
+    // if both classes are "final" (e.g. System.String[]) we can replace the comparison
+    // with `true/false` + null check.
+    if ((objCls != NO_CLASS_HANDLE) &&
+        // double check if it's final via impIsClassExact for arrays
+        (pIsExact || impIsClassExact(objCls)) && impIsClassExact(clsHnd))
```

Comment:
This can be just `(pIsExact || impIsClassExact(objCls))`.

We know exactly what type `clsHnd` refers to (modulo sharing).

Reply:
Ok, so I guess the `impIsClassExact(clsHnd)` is redundant? removed.

Reply:
Ok, so I guess the `impIsClassExact(clsHnd)` is redundant? removed.